### PR TITLE
Fix restore resolve indices logic to not include all templates

### PR DESCRIPTION
**Pull request by seut**
All templates must only be included if no concrete table is defined by the restore statement.

Follow up of https://github.com/crate/crate/commit/b0016d5.

### CODE REVIEW

**Comment by seut** [comment]
re-test this please

unrelated LR flaky test issue

File: server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java [hunk 1]
```diff
@@ -811,6 +811,28 @@ public void test_restore_empty_partitioned_table_does_not_restore_other_tables()
             "empty_parted2");
     }
 
+    @Test
+    public void test_restore_table_does_not_restore_any_partitioned_table() {
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:815]:
```suggestion
    @Test
    public void test_restore_table_does_not_restore_any_empty_partitioned_table() {
```

Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:815]: afaik non-empty partitioned tables are not affected by this issue?

If so, let's reflect it in the test name

**Review by BaurzhanSakhariev — APPROVED** [review]
Thanks! 

Added a minor comment

File: server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java (hunk 1 above)
Comment (seut) [reply server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:815]: Unfortunately not, it will always restore the template. The partitioned table is restored as empty as only the template is restored, I will re-phrase the changes entry.

Comment (seut) [reply server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:814]: I've rephrased the changes and also adjusted the test case to cover both scenarios, empty and non-empty partitioned tables. Please check,

Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:814]: Sorry for the late reply, got distracted a bit.

Have a minor comment on change entry, otherwise LGTM, thanks!

>Fixed a regression introduced with :ref:`version_5.6.0` that caused any
  partitioned table contained inside the snapshot to be restored, if not exists,
  by the :ref:`sql-restore-snapshot` statement 
  
`  if not exists` sounds confusing, can we drop it given that below we clarify that only definition was restored

**Review by BaurzhanSakhariev — APPROVED** [review]

File: server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java (hunk 1 above)
Comment (seut) [reply server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:814]: `if not exists` is even a SQL clause. That the definition only was restored is not related, it will be only restored if not exists.
